### PR TITLE
server: fix panic after delete store.

### LINKED ISSUE #563: pd-server response with panic message.
It happens when I send `store x` after `store delete x` using pd-ctl:
```
Failed to get store: [500] PANIC: runtime error: invalid memory address or nil pointer dereference
goroutine 4756617 [running]:
github.com/pingcap/pd/vendor/github.com/urfave/negroni.(*Recovery).ServeHTTP.func1(0x7ff975dfc140, 0xc434a64740, 0xc420251e00)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/vendor/github.com/urfave/negroni/recovery.go:41 +0x150
panic(0xdf7780, 0xc420014060)
        /usr/local/go/src/runtime/panic.go:458 +0x243
github.com/pingcap/pd/server/api.newStoreInfo(0xc4351346e0, 0xc4327d3860, 0xc4351346e0)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/server/api/store.go:63 +0x136
github.com/pingcap/pd/server/api.(*storeHandler).Get(0xc4201afde0, 0x7ff975dfc140, 0xc434a64780, 0xc42b9bc690)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/server/api/store.go:117 +0x241
github.com/pingcap/pd/server/api.(*storeHandler).Get-fm(0x7ff975dfc140, 0xc434a64780, 0xc42b9bc690)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/server/api/router.go:57 +0x48
net/http.HandlerFunc.ServeHTTP(0xc4201afdf0, 0x7ff975dfc140, 0xc434a64780, 0xc42b9bc690)
        /usr/local/go/src/net/http/server.go:1726 +0x44
github.com/pingcap/pd/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc4202626e0, 0x7ff975dfc140, 0xc434a64780, 0xc42b9bc690)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/vendor/github.com/gorilla/mux/mux.go:103 +0x255
github.com/pingcap/pd/vendor/github.com/urfave/negroni.Wrap.func1(0x7ff975dfc140, 0xc434a64780, 0xc42b9bc690, 0xc4345ed600)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/vendor/github.com/urfave/negroni/negroni.go:41 +0x4d
github.com/pingcap/pd/vendor/github.com/urfave/negroni.HandlerFunc.ServeHTTP(0xc4202fbba0, 0x7ff975dfc140, 0xc434a64780, 0xc42b9bc690, 0xc4345ed600)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/vendor/github.com/urfave/negroni/negroni.go:24 +0x4e
github.com/pingcap/pd/vendor/github.com/urfave/negroni.middleware.ServeHTTP(0x143d3a0, 0xc4202fbba0, 0xc4202fbc00, 0x7ff975dfc140, 0xc434a64780, 0xc42b9bc690)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/vendor/github.com/urfave/negroni/negroni.go:33 +0xa7
github.com/pingcap/pd/vendor/github.com/urfave/negroni.(middleware).ServeHTTP-fm(0x7ff975dfc140, 0xc434a64780, 0xc42b9bc690)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/vendor/github.com/urfave/negroni/negroni.go:33 +0x60
github.com/pingcap/pd/server/api.(*redirector).ServeHTTP(0xc420100270, 0x7ff975dfc140, 0xc434a64780, 0xc42b9bc690, 0xc4345ed5e0)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/server/api/redirector.go:44 +0x3a3
github.com/pingcap/pd/vendor/github.com/urfave/negroni.middleware.ServeHTTP(0x1439920, 0xc420100270, 0xc4202fbbe0, 0x7ff975dfc140, 0xc434a64780, 0xc42b9bc690)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/vendor/github.com/urfave/negroni/negroni.go:33 +0xa7
github.com/pingcap/pd/vendor/github.com/urfave/negroni.(*Negroni).ServeHTTP(0xc4203155f0, 0x7ff975dfc140, 0xc434a64740, 0xc42b9bc690)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/vendor/github.com/urfave/negroni/negroni.go:73 +0xdd
github.com/pingcap/pd/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc420262500, 0x7ff975dfc140, 0xc434a64740, 0xc42b9bc690)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/vendor/github.com/gorilla/mux/mux.go:103 +0x255
github.com/pingcap/pd/vendor/github.com/urfave/negroni.Wrap.func1(0x7ff975dfc140, 0xc434a64740, 0xc42b9bc690, 0xc4345ed5a0)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/vendor/github.com/urfave/negroni/negroni.go:41 +0x4d
github.com/pingcap/pd/vendor/github.com/urfave/negroni.HandlerFunc.ServeHTTP(0xc4202fbc40, 0x7ff975dfc140, 0xc434a64740, 0xc42b9bc690, 0xc4345ed5a0)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/vendor/github.com/urfave/negroni/negroni.go:24 +0x4e
github.com/pingcap/pd/vendor/github.com/urfave/negroni.middleware.ServeHTTP(0x143d3a0, 0xc4202fbc40, 0xc4202fbca0, 0x7ff975dfc140, 0xc434a64740, 0xc42b9bc690)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/vendor/github.com/urfave/negroni/negroni.go:33 +0xa7
github.com/pingcap/pd/vendor/github.com/urfave/negroni.(middleware).ServeHTTP-fm(0x7ff975dfc140, 0xc434a64740, 0xc42b9bc690)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/vendor/github.com/urfave/negroni/negroni.go:33 +0x60
github.com/pingcap/pd/vendor/github.com/urfave/negroni.(*Static).ServeHTTP(0xc420251e60, 0x7ff975dfc140, 0xc434a64740, 0xc42b9bc690, 0xc4345ed580)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/vendor/github.com/urfave/negroni/static.go:41 +0x7bf
github.com/pingcap/pd/vendor/github.com/urfave/negroni.middleware.ServeHTTP(0x143afa0, 0xc420251e60, 0xc4202fbc80, 0x7ff975dfc140, 0xc434a64740, 0xc42b9bc690)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/vendor/github.com/urfave/negroni/negroni.go:33 +0xa7
github.com/pingcap/pd/vendor/github.com/urfave/negroni.(middleware).ServeHTTP-fm(0x7ff975dfc140, 0xc434a64740, 0xc42b9bc690)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/vendor/github.com/urfave/negroni/negroni.go:33 +0x60
github.com/pingcap/pd/vendor/github.com/urfave/negroni.(*Recovery).ServeHTTP(0xc420251e00, 0x7ff975dfc140, 0xc434a64740, 0xc42b9bc690, 0xc4345ed560)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/vendor/github.com/urfave/negroni/recovery.go:64 +0x82
github.com/pingcap/pd/vendor/github.com/urfave/negroni.middleware.ServeHTTP(0x143af60, 0xc420251e00, 0xc4202fbc60, 0x7ff975dfc140, 0xc434a64740, 0xc42b9bc690)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/vendor/github.com/urfave/negroni/negroni.go:33 +0xa7
github.com/pingcap/pd/vendor/github.com/urfave/negroni.(*Negroni).ServeHTTP(0xc420251dd0, 0x1447520, 0xc42b1ddc70, 0xc42b9bc690)
        /home/jenkins/workspace/PD_POST_COMMIT_FLOW/go/src/github.com/pingcap/pd/vendor/github.com/urfave/negroni/negroni.go:73 +0xdd
net/http.(*ServeMux).ServeHTTP(0xc4233e23c0, 0x1447520, 0xc42b1ddc70, 0xc42b9bc690)
        /usr/local/go/src/net/http/server.go:2022 +0x7f
net/http.serverHandler.ServeHTTP(0xc42600f280, 0x1447520, 0xc42b1ddc70, 0xc42b9bc690)
        /usr/local/go/src/net/http/server.go:2202 +0x7d
net/http.(*conn).serve(0xc42283a380, 0x1448960, 0xc434934f00)
        /usr/local/go/src/net/http/server.go:1579 +0x4b7
created by net/http.(*Server).Serve
        /usr/local/go/src/net/http/server.go:2293 +0x44d
```

### PR DESCRIPTION
**Pull request by disksing**
Fix #563 

Using `new(StoreStatus)` causes the inner filed unset.

### CODE REVIEW

File: server/cluster.go [hunk 1]
```diff
@@ -401,7 +401,7 @@ func (c *RaftCluster) BuryStore(storeID uint64, force bool) error {
 	}
 
 	store.State = metapb.StoreState_Tombstone
-	store.stats = new(StoreStatus)
+	store.stats = newStoreStatus()
```
Comment (ngaut) [inline server/cluster.go]: The two new staff are very similar,  can we delete the newStoreStatus function?

Comment (disksing) [inline server/cluster.go]: `StoreStatus` has pointer field need to be initialized.

Comment (siddontang) [inline server/cluster.go]: Ping @disksing

**Comment by siddontang** [comment]
LGTM

**Pull request merged**